src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPEFileSaveWatcher.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.VS.Automation;
+using VSLangProj;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.TempPE
+{
+    /// <summary>
+    /// This class handles being notified whenever a source file is saved by the hierarchy
+    /// </summary>
+    [Export(typeof(IFileActionHandler))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    internal class TempPEFileSaveWatcher : IFileActionHandler
+    {
+        private readonly VSBuildManager _buildManager;
+        private readonly Lazy<ITempPEBuildManager> _tempPEBuildManager;
+
+        [ImportingConstructor]
+        public TempPEFileSaveWatcher(
+            [Import(typeof(BuildManager))]VSBuildManager buildManager,
```

Comment:
Again this class says "I only have a dependency on BuildManager" but it really has a dependency on VSBuildManager.

Reply:
This is related to https://github.com/dotnet/project-system/pull/4349#discussion_r239280321 and will be removed with that change.